### PR TITLE
REF cursor support

### LINKED ISSUE #58: Ref cursor support implementation
There are a couple of missing features for ref cursor support.

- Get data for `DPI_NATIVE_TYPE_STMT`
- `stmt_getNumQueryColumns` support.

To help with the testing the following snippet can be used to extract the ref cursor reference:

```
    #{var := Var, data := [Data]} = dpi:conn_newVar(Conn, Type, 'DPI_NATIVE_TYPE_STMT', 1, 0, false, false, undefined),
    dpi:stmt_bindByName(Stmt, Name, Var),
    RefCursor = dpi:data_get(Data)
    case dpi:stmt_fetch(RefCursor) of
            #{found := true} ->
                NumCols = dpi:stmt_getNumQueryColumns(RefCursor),
                {true, get_column_values(RefCursor, 1, NumCols)};
            #{found := false} ->
                {false, []}
        end
```

And the helper function to get the values:
```
get_column_values(_Stmt, ColIdx, Limit) when ColIdx > Limit -> [];
get_column_values(Stmt, ColIdx, Limit) ->
    #{data := Data} = dpi:stmt_getQueryValue(Stmt, ColIdx),
    [dpi:data_get(Data) | get_column_values(Stmt, ColIdx + 1, Limit)].
```

### PR DESCRIPTION
**Pull request by acautin**
fixes #58

Added missing implementation of `DPI_NATIVE_TYPE_STMT` to `data_get` function needed to extract ref cursor reference.

### CODE REVIEW

File: c_src/dpi_nif.h [hunk 1]
```diff
@@ -76,14 +76,24 @@ extern ERL_NIF_TERM ATOM_FALSE;
 extern ERL_NIF_TERM ATOM_ERROR;
 extern ERL_NIF_TERM ATOM_ENOMEM;
 
+#define DEBUG_DEACTIVATE_IOB
```
Comment (c-bik) [inline c_src/dpi_nif.h]: Please remove this macro and all code that activates with it

File: test/odpi_eunit.erl [hunk 2]
```diff
@@ -916,6 +916,50 @@ distributed([_, _]) ->
...
+    ?assertEqual(1, dpi:stmt_getNumQueryColumns(Stmt)),
+    dpi:stmt_release(Stmt),
+
+  
+    Stmt2 = dpi:conn_prepareStmt(Conn, false, <<"select 1, 2, 3, 4, 5 from dual">>, <<"">>),
+    5 = dpi:stmt_execute(Stmt2, []),
+    ?assertEqual(5, dpi:stmt_getNumQueryColumns(Stmt2)),
+    dpi:stmt_release(Stmt2),
+
+    ?_assert(true).
```
Comment (c-bik) [inline test/odpi_eunit.erl]: remove empty lines

File: test/odpi_eunit.erl [hunk 3]
```diff
@@ -916,6 +916,50 @@ distributed([_, _]) ->
...
+            ?debugFmt("Error: ~p", [R]),
+            R
+    end,
+  
+    %Stmt2 = dpi:conn_prepareStmt(Conn, false, <<"select 1, 2, 3, 4, 5 from dual">>, <<"">>),
+    %5 = dpi:stmt_execute(Stmt2, []),
+    %?assertEqual(5, dpi:stmt_getNumQueryColumns(Stmt2)),
+    %dpi:stmt_release(Stmt2),
+
+    ?_assert(true).
```
Comment (c-bik) [inline test/odpi_eunit.erl]: Complete the test with reading from REF cursor

File: test/odpi_eunit.erl [hunk 4]
```diff
@@ -916,6 +916,50 @@ distributed([_, _]) ->
...
+    5 = dpi:stmt_execute(Stmt2, []),
+    ?assertEqual(5, dpi:stmt_getNumQueryColumns(Stmt2)),
+    dpi:stmt_release(Stmt2),
+
+    ?_assert(true).
+
+ref_cursor([Context, Conn]) -> 
+    try
+    ?debugFmt("Pass: ~p", [-1]),
+    Stmt = dpi:conn_prepareStmt(Conn, false, <<"create or replace procedure FOOBAR() is begin select 123 from dual; end FOOBAR" >>, <<"">>),
```
Comment (c-bik) [inline test/odpi_eunit.erl]: Multiline the SQL for readablity
```erlang
<<
    "create or replace ..."
    ...
    "end FOOBAR"
>>
```

Comment (acautin) [inline test/odpi_eunit.erl]: here passing the ref_cursor as an out variable is needed.
```
create or replace procedure FOOBAR(p_cur out sys_refcursor)
    is begin 
         open p_cur for 'select 123 from dual;
    end FOOBAR
```

**Pull request merged**